src/app/cluster/details/cluster/edit-cluster/component.ts

```diff
@@ -197,6 +210,57 @@ export class EditClusterComponent implements OnInit, OnDestroy {
       : AuditPolicyPreset.Custom;
   }
 
+  onLabelsChange(labels: Record<string, string>): void {
+    this.labels = labels;
+  }
+
+  onPodNodeSelectorAdmissionPluginConfigChange(config: Record<string, string>): void {
+    this.podNodeSelectorAdmissionPluginConfig = config;
+
+    this.handleClusterDefaultNodeSelector(config);
+  }
+
+  handleClusterDefaultNodeSelector(nodeConfig: Record<string, string>): void {
+    let labelEntries: [string, string][] = (this.form.get(Controls.Labels).value?.labels ?? [])
+      .filter(({key, value}: KeyValue<string, string>) => Boolean(key && value))
+      .map(({key, value}: KeyValue<string, string>) => [key, value]);
+
+    const [_ = '', labelSelector = ''] =
+      Object.entries(nodeConfig).find(
+        ([key, _]) => key.toLowerCase() === this.CLUSTER_DEFAULT_NODE_SELECTOR_NAMESPACE.toLowerCase()
+      ) ?? [];
+
+    const [currKey = '', currValue = ''] = labelSelector.split('=') ?? [];
+
+    const hasCurrLabel = Boolean(currKey && currValue);
```

Comment:
`hasCurrLabel` at first look i thought it was a function checking from existing labels [] if current key/value already existed in array but if we are checking both key & value exists can we rename variable?

e.g keyValueExists?  or any better name? wdyt?